src/clientHandler.ts

```diff
@@ -1,168 +1,53 @@
-import * as vscode from 'vscode';
-import TelemetryReporter from '@vscode/extension-telemetry';
-import {
-  DocumentSelector,
-  Executable,
-  LanguageClient,
-  LanguageClientOptions,
-  RevealOutputChannelOn,
-  ServerOptions,
-  State,
-} from 'vscode-languageclient/node';
+import { Executable, ServerOptions } from 'vscode-languageclient/node';
 import { config } from './utils/vscode';
 import { ServerPath } from './utils/serverPath';
-import { PartialManifest, CustomSemanticTokens } from './features/semanticTokens';
-import { ShowReferencesFeature } from './features/showReferences';
-import { TelemetryFeature } from './features/telemetry';
 
-/**
- * ClientHandler maintains lifecycles of language clients
- * based on the server's capabilities
- */
-export class ClientHandler {
-  private client: LanguageClient | undefined;
-  private commands: string[] = [];
+export async function getServerOptions(lsPath: ServerPath): Promise<ServerOptions> {
+  const cmd = await lsPath.resolvedPathToBinary();
```

Comment:
This function has little reason to exist outside of handling LS paths and args. When we address #791 we can fold this into the main client activation